addons/web/views/report_templates.xml

```diff
@@ -66,13 +61,244 @@
         </html>
     </template>
 
+    <template id="report_preview_layout" name="Report layout">&lt;!DOCTYPE html&gt;
+        <html t-att-lang="lang and lang.replace('_', '-')"
+              t-att-data-report-margin-top="data_report_margin_top"
+              t-att-data-report-header-spacing="data_report_header_spacing"
+              t-att-data-report-dpi="data_report_dpi"
+              t-att-data-report-landscape="data_report_landscape"
+              t-att-web-base-url="web_base_url"
+        >
+            <head>
+                <meta charset="utf-8"/>
+                <meta name="viewport" content="initial-scale=1"/>
+                <title><t t-esc="title or 'Odoo Report'"/></title>
+                <t t-call-assets="web.report_assets_common" t-js="false"/>
+                <t t-call-assets="web.assets_common" t-css="false"/>
+                <style>
+                    <t t-raw="preview_css"/>
+
+                    /**
+                        Some css is overridden as it doesn't work properly in the preview.
+                    **/
+
+                    .o_background_footer, .o_standard_footer, .o_boxed_footer, .o_clean_footer {
+                        margin-top: 50px !important;
+                    }
+                    .o_background_header {
+                        min-width: initial !important;
+                    }
+                </style>
+            </head>
+            <body t-att-class="'container' if not full_width else 'container-fluid'">
+                <div id="wrapwrap">
+                        <t t-raw="0"/>
+                </div>
+            </body>
+        </html>
+    </template>
+
     <template id="html_container">
         <t t-set="body_classname" t-value="'container'"/>
         <t t-call="web.report_layout">
             <t t-raw="0"/>
         </t>
     </template>
 
+    <template id="html_preview_container">
+        <t t-set="body_classname" t-value="'container'"/>
+            <t t-call="web.report_preview_layout">
+                <t t-raw="0"/>
+            </t>
+    </template>
+
```

Comment:
There are quite a lot of empty lines in this diff; in general we only use them to separate different templates